serverless.yaml

```diff
@@ -1,362 +1,417 @@
+      Resource:
+        - !GetAtt S3KMSKey.Arn
+        - !GetAtt DynamodbKMSKey.Arn
+        - !GetAtt ElasticSearchKMSKey.Arn
+    - Action:
+        - 'dynamodb:Query'
+        - 'dynamodb:Scan'
+        - 'dynamodb:GetItem'
+        - 'dynamodb:PutItem'
+        - 'dynamodb:UpdateItem'
+        - 'dynamodb:DeleteItem'
+        - 'dynamodb:BatchWriteItem'
+      Effect: Allow
+      Resource:
+        - !GetAtt ResourceDynamoDBTableV2.Arn
+    - Action:
+        - 'es:*'
+      Effect: Allow
+      Resource:
+        - !Join ['', [!GetAtt ElasticSearchDomain.Arn, '/*']]
+    - Action:
+        - 's3:*'
+      Effect: Allow
+      Resource:
+        - !GetAtt FHIRBinaryBucket.Arn
+        - !Join ['', [!GetAtt FHIRBinaryBucket.Arn, '/*']]
+    - Action:
+        - 'xray:PutTraceSegments'
+        - 'xray:PutTelemetryRecords'
+      Effect: Allow
+      Resource:
+        - '*'
+  variableSyntax: "\\${((?!AWS)[ ~:a-zA-Z0-9._@'\",\\-\\/\\(\\)]+?)}" # Use this for allowing CloudFormation Pseudo-Parameters in your serverless.yml
+  logs:
+    restApi:
+      executionLogging: true
+      fullExecutionData: true
+  tracing:
+    apiGateway: true
+
+functions:
+  fhirServer:
+    timeout: 40
+    description: 'FHIR API Server'
+    events:
+      - http:
+          authorizer:
+            type: COGNITO_USER_POOLS
+            authorizerId:
+              Ref: ApiGatewayAuthorizer
+            scopes: # must have both scopes
+              - 'openid'
+              - 'profile'
+              - 'aws.cognito.signin.user.admin'
+          method: ANY
+          path: '/{proxy+}'
+          private: true
+    handler: src/index.default
+    provisionedConcurrency: 5
+
+  ddbToEs:
+    timeout: 300
+    runtime: nodejs12.x
+    description: 'Write DDB changes from `resource` table to ElasticSearch service'
+    role: DdbToEsLambdaRole
+    handler: ddbToEsLambda/index.handler
+    events:
+      - stream:
+          type: dynamodb
+          arn:
+            'Fn::GetAtt':
+              - ResourceDynamoDBTableV2
+              - StreamArn
+          batchSize: 100
+          maximumRetryAttempts: 3
+          startingPosition: LATEST
+
+resources:
+  - Description: (SO0128) - Solution - Master Template - This template creates all the necessary resources to deploy FHIR Works on AWS; a framework to deploy a FHIR server on AWS.
+  - Parameters:
+      ShouldCreateManagementUser:
+        Type: String
+        Default: true
+        Description: "Set to 'true' to create management IAM user, 'false' if providing user ARN. Default: true"
+        AllowedValues: [ true, false ]
+      UserPassword:
+        Type: String
+        Description: Password for development IAM user. Ignored if ShouldCreateManagementUser is false.
+        NoEcho: true
+      ExistingManagementUserARN:
+        Type: String
+        Default: ""
+        Description: The ARN of the IAM user to use for KMS key management. Ignored if ShouldCreateManagementUser is true.
```

Comment:
When running `sls offline` to allow local debugging of the FHIR server, serverless uses the credentials of this user. ES is set to allow this user to make requests to ES.
https://github.com/awslabs/fhir-works-on-aws-deployment/blob/5780f97afe427b26405d82062fbc86b7f61d43a8/cloudformation/elasticsearch.yaml#L149